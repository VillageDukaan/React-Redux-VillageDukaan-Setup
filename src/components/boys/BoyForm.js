import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
import SelectInput from "../common/SelectInput";

const BoyForm = ({
  boy,
  managers,
  onSave,
  onChange,
  saving = false,
  errors = {},
}) => {
  return (
    <form onSubmit={onSave}>
      <h2>{boy.id ? "Edit" : "Add"} Delivery Boy</h2>
      {errors.onSave && (
        <div className="alert alert-danger">{errors.onSave}</div>
      )}
      <TextInput
        name="name"
        label="Name"
        value={boy.name}
        onChange={onChange}
        error={errors.name}
      />
      <SelectInput
        name="managerId"
        label="Manager"
        value={boy.managerId || ""}
        defaultOption="Select Manager"
        options={managers.map((manager) => ({
          value: manager.id,
          text: manager.name,
        }))}
        onChange={onChange}
        error={errors.manager}
      />
      <TextInput
        name="category"
        label="Category"
        value={boy.category}
        onChange={onChange}
        error={errors.category}
      />
      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving..." : "Save"}
      </button>
    </form>
  );
};

BoyForm.propTypes = {
  managers: PropTypes.array.isRequired,
  boy: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool,
};

export default BoyForm;
