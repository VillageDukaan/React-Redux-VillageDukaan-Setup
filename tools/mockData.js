const deliveryBoys = [
  {
    id: 1,
    name: "Lova Katari",
    slug: "delivery-to-rajahmundry",
    managerId: 1,
    category: "Short DIstance Traveller",
  },
  {
    id: 2,
    name: "Phani",
    slug: "delivery-to-hyderabad",
    managerId: 1,
    category: "Short DIstance Traveller",
  },
  {
    id: 3,
    name: "Sagar",
    slug: "delivery-to-kakinada",
    managerId: 1,
    category: "Short DIstance Traveller",
  },
  {
    id: 4,
    name: "Afroz",
    slug: "delivery-to-kovvur",
    managerId: 1,
    category: "Short DIstance Traveller",
  },
  {
    id: 5,
    name: "Sai",
    slug: "delivery-to-nidadavole",
    managerId: 1,
    category: "Short DIstance Traveller",
  },
  {
    id: 6,
    name: "Manikanta",
    slug: "delivery-to-ramachandrapuram",
    managerId: 1,
    category: "Short DIstance Traveller",
  },
  {
    id: 7,
    name: "Raj Babu",
    slug: "delivery-to-rajanagaram",
    managerId: 1,
    category: "Long DIstance Traveller",
  },
  {
    id: 8,
    name: "Ramarao",
    slug: "delivery-to-kadiyam",
    managerId: 1,
    category: "Long DIstance Traveller",
  },
  {
    id: 9,
    name: "Salomi",
    slug: "delivery-to-bangalore",
    managerId: 1,
    category: "Long DIstance Traveller",
  },
  {
    id: 10,
    name: "Jaswanth",
    slug: "delivery-to-mumbai",
    managerId: 1,
    category: "Long DIstance Traveller",
  },
];

const managers = [
  { id: 1, name: "Lova" },
  { id: 2, name: "Lakshmi" },
  { id: 3, name: "Satyanarayana" },
];

const newDeliveryBoy = {
  id: null,
  name: "",
  managerId: null,
  category: "",
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newDeliveryBoy,
  deliveryBoys,
  managers,
};
