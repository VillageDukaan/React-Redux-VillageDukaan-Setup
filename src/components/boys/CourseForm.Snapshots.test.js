import React from "react";
import BoyForm from "./BoyForm";
import renderer from "react-test-renderer";
import { deliveryBoys, managers } from "../../../tools/mockData";

it("sets submit button label 'Saving... when saving is true", () => {
  const tree = renderer.create(
    <BoyForm
      boy={deliveryBoys[0]}
      managers={managers}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <BoyForm
      boy={deliveryBoys[0]}
      managers={managers}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
