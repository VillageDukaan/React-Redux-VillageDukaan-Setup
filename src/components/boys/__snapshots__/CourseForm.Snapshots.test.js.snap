// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Delivery Boy
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="Lova Katari"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="managerId"
    >
      Manager
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="managerId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Manager
        </option>
        <option
          value={1}
        >
          Lova
        </option>
        <option
          value={2}
        >
          Lakshmi
        </option>
        <option
          value={3}
        >
          Satyanarayana
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="Short DIstance Traveller"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving... when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Delivery Boy
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="Lova Katari"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="managerId"
    >
      Manager
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="managerId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Manager
        </option>
        <option
          value={1}
        >
          Lova
        </option>
        <option
          value={2}
        >
          Lakshmi
        </option>
        <option
          value={3}
        >
          Satyanarayana
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="Short DIstance Traveller"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
